sdk/core/core-http/lib/axiosHttpClient.ts

```diff
@@ -227,7 +227,8 @@ declare type ProxyAgent = { isHttps: boolean; agent: http.Agent | https.Agent };
 export function createProxyAgent(requestUrl: string, proxySettings: ProxySettings, headers?: HttpHeaders): ProxyAgent {
   const tunnelOptions: tunnel.HttpsOverHttpsOptions = {
     proxy: {
-      host: URLBuilder.parse(proxySettings.host).getHost(),
+      // TODO: Handle case where getHost() returns undefined
+      host: URLBuilder.parse(proxySettings.host).getHost()!,
       port: proxySettings.port,
       headers: (headers && headers.rawHeaders()) || {}
```

Comment:
@daviwil: Since `ProxyOptions.headers` now allows undefined, do you think we should make this change?  I assume the `|| {}` was to handle `ProxyOptions.headers` not allowing undefined.

```suggestion
       headers: headers && headers.rawHeaders()
```

Reply:
Looks like the tunnel API is basically [doing the same thing](https://github.com/koichik/node-tunnel/blob/master/lib/tunnel.js#L120) in this case, but perhaps its better to let `headers` be `undefined` so that `tunnel` can choose its own "default" behavior if they ever want to change it.